README.md

```diff
@@ -52,6 +52,43 @@ FeatureGateMyFeatureName = newFeatureGate("MyFeatureName").
 			mustRegister()
 ```
 
+### defining tests
+Tests are logically associated with FeatureGates.
+When adding any FeatureGated functionality a new test file is required.
+The test files are located in `<group>/<version>/tests/<crd-name>/FeatureGate.yaml`
+```
+route
+  v1
+    tests
+      routes.route.openshift.io
+        AAA_ungated.yaml
+        ExternalRouteCertificate.yaml
+```
+
+Here's an `AAA_ungated.yaml` example
+```yaml
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "Route"
+crdName: routes.route.openshift.io
+tests:
+```
+
+Here's an `ExternalRouteCertificate.yaml` example
+```yaml
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "Route"
+crdName: routes.route.openshift.io
+featureGate: ExternalRouteCertificate
+tests:
+```
+
+The integration tests use the crdName and featureGate to determine which tests apply to which manifests and automatically
+react to changes when the FeatureGates are enabled/disabled on various FeatureSets and ClusterProfiles.
+
+[gen-minimal-test.sh](tests%2Fhack%2Fgen-minimal-test.sh) can still function to stub out files if you don't want to
```

Comment:
```suggestion
[`gen-minimal-test.sh`](tests/hack/gen-minimal-test.sh) can still function to stub out files if you don't want to
```
...although surprisingly GitHub seems equally happy to serve up both https://github.com/openshift/api/blob/ca1cfcd7c0e515141b98205f394b7c3f006ecb40/tests/hack/gen-minimal-test.sh and https://github.com/openshift/api/blob/ca1cfcd7c0e515141b98205f394b7c3f006ecb40/tests%2Fhack%2Fgen-minimal-test.sh, so the percent-encoding is unnecessary but doesn't actually break the link.